src/XrdCrypto/XrdCryptosslgsiAux.cc

```diff
@@ -1141,6 +1146,9 @@ int XrdCryptosslX509SignProxyReq(XrdCryptoX509 *xcpi, XrdCryptoRSA *kcpi,
       return -kErrPX_Signing;
    }
 
+   EVP_PKEY_free( ekpi ); // decrement reference counter
+   X509_EXTENSION_free( ext );
```

Comment:
As above, the extension should be owned by the proxy certificate ...

Reply:
As above ;-) X509_add_ext() duplicates the object internally ;-)